### ISSUE TITLE
report function 3rd arguments should throw error

### ISSUE THREAD

**Issue by azu** [textlint/textlint#316]
```js
report(node,
  new RuleError(`using "please" in a set of instructions is overdoing the politeness.`), {
    index,
    fix: fixer.replaceTextRange(range, result.replace)
  }
)
```

It is a common mistake.
We should throw error asap.
